packages/common/src/failure.ts

```diff
@@ -329,15 +329,22 @@ export async function errorToFailure(err: unknown, dataConverter: DataConverter)
     source: FAILURE_SOURCE,
   };
 
-  if (err instanceof Error) {
-    return { ...base, message: err.message ?? '', stackTrace: cutoffStackTrace(err.stack) };
+  if (isRecord(err) && hasOwnProperties(err, ['message', 'stack'])) {
```

Comment:
Why not check `instanceof Error`?

Reply:
`err instanceof Error` is false when the Workflow does `throw new Error('unhandled rejection');`. Could also be helpful in the rare case of people throwing error-like objects

Reply:
Oh, that makes sense because the `Error` is constructed in isolate context and converted into failure outside of that. We should probably work around it and ask the workflow to convert to a failure since it has an instance of the user's data converter.

Reply:
By moving this `errorToFailure`:

https://github.com/temporalio/sdk-typescript/blob/0488e8a74fa78c44d9fcad6e2631c7aae868aae5/packages/worker/src/workflow/vm.ts#L196

to here, instead of throwing? 

https://github.com/temporalio/sdk-typescript/blob/0488e8a74fa78c44d9fcad6e2631c7aae868aae5/packages/workflow/src/internals.ts#L504